app/src/main/java/com/example/taskfour/viewModel/ListViewModel.kt

```diff
@@ -0,0 +1,63 @@
+package com.example.taskfour.viewModel
+
+import android.app.Application
+import android.widget.Toast
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.viewModelScope
+import com.example.taskfour.model.CryptoModel
+import com.example.taskfour.room.CryptoDatabase
+import com.example.taskfour.room.CryptoRepository
+import com.example.taskfour.service.CryptoApiService
+import kotlinx.coroutines.launch
+
+class ListViewModel(application: Application) : AndroidViewModel(application) {
+    var readAllData: LiveData<List<CryptoModel>>
+    var repostory: CryptoRepository
+
+    private val cryptoApiService = CryptoApiService()
+
+    private val cryptoList = MutableLiveData<List<CryptoModel>>()
+    val cryptoListObs: LiveData<List<CryptoModel>>
```

Comment:
Immutablity şu şekilde olması lazım;
```suggestion
    private val _cryptoList = MutableLiveData<List<CryptoModel>>()
    val cryptoList: LiveData<List<CryptoModel>>
```